src/commands/cmd_query.c

```diff
@@ -142,6 +142,11 @@ void Graph_Query(void *args) {
 	if(exec_type == EXECUTION_TYPE_INVALID) goto cleanup;
 
 	bool readonly = AST_ReadOnly(ast->root);
+	if(!readonly && strcasecmp(CommandCtx_GetCommandName(command_ctx), "graph.RO_QUERY") == 0) {
```

Comment:
Might be a bit cleaner if `strcasecmp(CommandCtx_GetCommandName(command_ctx), "graph.RO_QUERY") == 0`
was an inline function, e.g. `inline bool _readonly_cmd_mode(CommandCtx *ctx)`